adoc/extensions/sycl_khr_free_function_commands.adoc

```diff
@@ -0,0 +1,928 @@
+= SYCL_KHR_FREE_FUNCTION_COMMANDS
+
+This extension provides an alternative mechanism for submitting commands to a
+device via free-functions that require developers to opt-in to the creation of
+[code]#event# objects.
+
+== Dependencies
+
+This extension has no dependencies on other extensions.
+
+== Feature test macro
+
+An implementation supporting this extension must predefine the macro
+[code]#SYCL_KHR_FREE_FUNCTION_COMMANDS# to one of the values defined in the
+table below.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+== Usage example
+
+The example below rewrites the application from
+<<subsec:example.sycl.application>> to demonstrate the usage of this extension.
+
+[source,role=synopsis]
+----
+#include <iostream>
+#include <sycl/sycl.hpp>
+using namespace sycl; // (optional) avoids need for "sycl::" before SYCL names
+
+// Size of the matrices
+constexpr size_t N = 2000;
+constexpr size_t M = 3000;
+
+int main() {
+  // Create a queue to work on
+  queue myQueue;
+
+  // Create some 2D arrays of float for our matrices
+  float* a = malloc_shared<float>(N * M, myQueue);
+  float* b = malloc_shared<float>(N * M, myQueue);
+  float* c = malloc_shared<float>(N * M, myQueue);
+
+  // Launch an asynchronous kernel to initialize a
+  // Use khr::submit to get a handler, even though not required here
```

Comment:
What do you mean?

Reply:
The `handler` isn't strictly required here, because we don't use any functionality of the `handler` other than to enqueue the kernel.  But I agree the wording is confusing, so I've simplified it in ce08652764e915d0d5bb16628869a7e2f5efd657.  The wording is now more consistent with the usage below that doesn't use a `handler`.